SUS-271 Update phalanx_stats charset to match MW connections' default + minor cleanups

armonr opened the pull request:
This should fix phalanx logs going forward.
Also few minor changes added:
- ps_blocker_hit can be either null or zero, check for emptiness
- There are cases where phalanx stats index page does not set some of the variables expected in the template; don't operate on them if not set.

/CC @Grunny @mixth-sense

Commit:
SUS-271 NFC Normalize the name of user being blocked prior to phalanx log
This should fix the issue where the usernames contains Russian or other combo characters; they get normalized before being written to mysql or scribe

Commit:
SUS-271 Few safety checks on phalanx logs
1. ps_blocker_hit can be either null or zero, check for emptiness
2. There are cases where phalanx stats index page does not set some of the variables expected in the template; don't operate on them if not set.

WikiaTech commented:
SonarQube analysis reported 26 issues
* ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) 6 major
* ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) 8 minor
* ![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) 12 info

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [Phalanx.class.php#L77](https://github.com/Wikia/app/blob/6d534081c24792b2cfa454504fc99387fa2ca5ce/extensions/wikia/PhalanxII/fallback/Phalanx.class.php#L77): Add curly braces around the nested statement(s). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS121)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [Phalanx.class.php#L153](https://github.com/Wikia/app/blob/6d534081c24792b2cfa454504fc99387fa2ca5ce/extensions/wikia/PhalanxII/fallback/Phalanx.class.php#L153): The Cyclomatic Complexity of this function "getFromFilterShort" is 21 which is greater than 20 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1541)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [Phalanx.class.php#L406](https://github.com/Wikia/app/blob/6d534081c24792b2cfa454504fc99387fa2ca5ce/extensions/wikia/PhalanxII/fallback/Phalanx.class.php#L406): The Cyclomatic Complexity of this function "findBlocked" is 22 which is greater than 20 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1541)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [Phalanx.class.php#L457](https://github.com/Wikia/app/blob/6d534081c24792b2cfa454504fc99387fa2ca5ce/extensions/wikia/PhalanxII/fallback/Phalanx.class.php#L457): Refactor this code to not nest more than 4 "if", "for", "while", "switch" and "try" statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS134)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [Phalanx.class.php#L472](https://github.com/Wikia/app/blob/6d534081c24792b2cfa454504fc99387fa2ca5ce/extensions/wikia/PhalanxII/fallback/Phalanx.class.php#L472): Refactor this code to not nest more than 4 "if", "for", "while", "switch" and "try" statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS134)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [Phalanx.class.php#L483](https://github.com/Wikia/app/blob/6d534081c24792b2cfa454504fc99387fa2ca5ce/extensions/wikia/PhalanxII/fallback/Phalanx.class.php#L483): Refactor this code to not nest more than 4 "if", "for", "while", "switch" and "try" statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS134)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [PhalanxStatsPager.class.php#L68](https://github.com/Wikia/app/blob/6d534081c24792b2cfa454504fc99387fa2ca5ce/extensions/wikia/PhalanxII/classes/PhalanxStatsPager.class.php#L68): Immediately return this expression instead of assigning it to the temporary variable "$html". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1488)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [Phalanx.class.php#L77](https://github.com/Wikia/app/blob/6d534081c24792b2cfa454504fc99387fa2ca5ce/extensions/wikia/PhalanxII/fallback/Phalanx.class.php#L77): 2 statements were found on this line. Reformat the code to have only one statement per line. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS122)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [Phalanx.class.php#L100](https://github.com/Wikia/app/blob/6d534081c24792b2cfa454504fc99387fa2ca5ce/extensions/wikia/PhalanxII/fallback/Phalanx.class.php#L100): Define a constant instead of duplicating this literal "closestExpire" 8 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [Phalanx.class.php#L117](https://github.com/Wikia/app/blob/6d534081c24792b2cfa454504fc99387fa2ca5ce/extensions/wikia/PhalanxII/fallback/Phalanx.class.php#L117): Define a constant instead of duplicating this literal "phalanx" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)

mszabo-wikia commented:
@armonr In Phalanx scala service we perform stats logging via scribe: https://github.com/Wikia/scala-backend/blob/master/phalanx/src/main/scala/com/wikia/phalanx/service.scala#L133

Does this fix stats entries generated that way?

Commit:
SUS-271 Script to convert phalanx_stats charset

Commit:
Minor change to PhalanxStatsPager::formatRow & Revert normalizing
Reverting because once database connection charset matches the table charset, normalization should not be needed

WikiaTech commented on maintenance/wikia/convertPhalanxStatsCharset.php:
```diff
@@ -0,0 +1,46 @@
+$alterCharset = "ALTER TABLE $table CONVERT TO CHARACTER SET latin1;";
+
+if ( isset( $options['verbose'] ) ) {
+	echo( "Executing query on $schema.$table: $alterCharset ...\n" );
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove the parentheses from this "echo" call. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2041)

WikiaTech commented on extensions/wikia/PhalanxII/classes/PhalanxStatsPager.class.php:
```diff
@@ -45,26 +45,34 @@ function getPagingQueries() {
 		$html  = Html::openElement( 'li' );
-		$html .= $this->msg( 'phalanx-stats-row', $type, $username, $url, $timestamp )->parse();
+		$html .= $this->msg( 'phalanx-stats-row', $type, $specialContributionsURL, $url, $timestamp )->parse();
 		$html .= Html::closeElement( 'li' );
```
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Immediately return this expression instead of assigning it to the temporary variable "$html". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1488)

WikiaTech commented:
SonarQube analysis reported 27 issues
* ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) 7 major
* ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) 8 minor
* ![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) 12 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [Phalanx.class.php#L77](https://github.com/Wikia/app/blob/f0538db58e8a07e39659f8486a12624a5f851c6f/extensions/wikia/PhalanxII/fallback/Phalanx.class.php#L77): Add curly braces around the nested statement(s). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS121)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [Phalanx.class.php#L153](https://github.com/Wikia/app/blob/f0538db58e8a07e39659f8486a12624a5f851c6f/extensions/wikia/PhalanxII/fallback/Phalanx.class.php#L153): The Cyclomatic Complexity of this function "getFromFilterShort" is 21 which is greater than 20 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1541)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [Phalanx.class.php#L406](https://github.com/Wikia/app/blob/f0538db58e8a07e39659f8486a12624a5f851c6f/extensions/wikia/PhalanxII/fallback/Phalanx.class.php#L406): The Cyclomatic Complexity of this function "findBlocked" is 22 which is greater than 20 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1541)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [Phalanx.class.php#L457](https://github.com/Wikia/app/blob/f0538db58e8a07e39659f8486a12624a5f851c6f/extensions/wikia/PhalanxII/fallback/Phalanx.class.php#L457): Refactor this code to not nest more than 4 "if", "for", "while", "switch" and "try" statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS134)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [Phalanx.class.php#L472](https://github.com/Wikia/app/blob/f0538db58e8a07e39659f8486a12624a5f851c6f/extensions/wikia/PhalanxII/fallback/Phalanx.class.php#L472): Refactor this code to not nest more than 4 "if", "for", "while", "switch" and "try" statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS134)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [Phalanx.class.php#L483](https://github.com/Wikia/app/blob/f0538db58e8a07e39659f8486a12624a5f851c6f/extensions/wikia/PhalanxII/fallback/Phalanx.class.php#L483): Refactor this code to not nest more than 4 "if", "for", "while", "switch" and "try" statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS134)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [Phalanx.class.php#L77](https://github.com/Wikia/app/blob/f0538db58e8a07e39659f8486a12624a5f851c6f/extensions/wikia/PhalanxII/fallback/Phalanx.class.php#L77): 2 statements were found on this line. Reformat the code to have only one statement per line. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS122)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [Phalanx.class.php#L100](https://github.com/Wikia/app/blob/f0538db58e8a07e39659f8486a12624a5f851c6f/extensions/wikia/PhalanxII/fallback/Phalanx.class.php#L100): Define a constant instead of duplicating this literal "closestExpire" 8 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [Phalanx.class.php#L117](https://github.com/Wikia/app/blob/f0538db58e8a07e39659f8486a12624a5f851c6f/extensions/wikia/PhalanxII/fallback/Phalanx.class.php#L117): Define a constant instead of duplicating this literal "phalanx" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [Phalanx.class.php#L132](https://github.com/Wikia/app/blob/f0538db58e8a07e39659f8486a12624a5f851c6f/extensions/wikia/PhalanxII/fallback/Phalanx.class.php#L132): Define a constant instead of duplicating this literal "exact" 5 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)

armonr commented:
@TK-999 Good point. I did however find another issue which seems to be the root cause of the encoding issue and that is the the table holding phalanx block logs is in utf8 charset whereas virtually all the connections made in app are set to latin1. And so there should be no changes required in the service.

macbre commented on maintenance/wikia/convertPhalanxStatsCharset.php:
```diff
@@ -0,0 +1,46 @@
+	wfWaitForSlaves();
+
+	$dbw->query( $alterCharset );
+	$dbw->commit();
```
`commit` is not really needed here. MW does not start a transaction when running in CLI mode.

macbre replied:
Creating a maintenance script for running a one-time DB query is kind of an overkill :smile:

armonr replied:
@macbre I was not sure how it is done in the app repo 😸 I figured at least the change gets code reviewed before being attempted.

armonr committed:
Address CR - Remove commit

/CC @macbre

WikiaTech commented:
SonarQube analysis reported 26 issues
* ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) 6 major
* ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) 8 minor
* ![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) 12 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [Phalanx.class.php#L77](https://github.com/Wikia/app/blob/f67fc18d493e00e39b32dfbf2367f2323c175a3e/extensions/wikia/PhalanxII/fallback/Phalanx.class.php#L77): Add curly braces around the nested statement(s). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS121)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [Phalanx.class.php#L153](https://github.com/Wikia/app/blob/f67fc18d493e00e39b32dfbf2367f2323c175a3e/extensions/wikia/PhalanxII/fallback/Phalanx.class.php#L153): The Cyclomatic Complexity of this function "getFromFilterShort" is 21 which is greater than 20 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1541)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [Phalanx.class.php#L406](https://github.com/Wikia/app/blob/f67fc18d493e00e39b32dfbf2367f2323c175a3e/extensions/wikia/PhalanxII/fallback/Phalanx.class.php#L406): The Cyclomatic Complexity of this function "findBlocked" is 22 which is greater than 20 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1541)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [Phalanx.class.php#L457](https://github.com/Wikia/app/blob/f67fc18d493e00e39b32dfbf2367f2323c175a3e/extensions/wikia/PhalanxII/fallback/Phalanx.class.php#L457): Refactor this code to not nest more than 4 "if", "for", "while", "switch" and "try" statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS134)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [Phalanx.class.php#L472](https://github.com/Wikia/app/blob/f67fc18d493e00e39b32dfbf2367f2323c175a3e/extensions/wikia/PhalanxII/fallback/Phalanx.class.php#L472): Refactor this code to not nest more than 4 "if", "for", "while", "switch" and "try" statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS134)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [Phalanx.class.php#L483](https://github.com/Wikia/app/blob/f67fc18d493e00e39b32dfbf2367f2323c175a3e/extensions/wikia/PhalanxII/fallback/Phalanx.class.php#L483): Refactor this code to not nest more than 4 "if", "for", "while", "switch" and "try" statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS134)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [Phalanx.class.php#L77](https://github.com/Wikia/app/blob/f67fc18d493e00e39b32dfbf2367f2323c175a3e/extensions/wikia/PhalanxII/fallback/Phalanx.class.php#L77): 2 statements were found on this line. Reformat the code to have only one statement per line. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS122)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [Phalanx.class.php#L100](https://github.com/Wikia/app/blob/f67fc18d493e00e39b32dfbf2367f2323c175a3e/extensions/wikia/PhalanxII/fallback/Phalanx.class.php#L100): Define a constant instead of duplicating this literal "closestExpire" 8 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [Phalanx.class.php#L117](https://github.com/Wikia/app/blob/f67fc18d493e00e39b32dfbf2367f2323c175a3e/extensions/wikia/PhalanxII/fallback/Phalanx.class.php#L117): Define a constant instead of duplicating this literal "phalanx" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [Phalanx.class.php#L132](https://github.com/Wikia/app/blob/f67fc18d493e00e39b32dfbf2367f2323c175a3e/extensions/wikia/PhalanxII/fallback/Phalanx.class.php#L132): Define a constant instead of duplicating this literal "exact" 5 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)

Commit:
Merge branch 'dev' into SUS-271

 Conflicts:
	extensions/wikia/PhalanxII/templates/PhalanxStatsSpecial_index.php

WikiaTech commented:
SonarQube analysis reported 26 issues
* ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) 6 major
* ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) 8 minor
* ![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) 12 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [Phalanx.class.php#L77](https://github.com/Wikia/app/blob/76ef81c3382df88f8708e5748c551f55ef556204/extensions/wikia/PhalanxII/fallback/Phalanx.class.php#L77): Add curly braces around the nested statement(s). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS121)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [Phalanx.class.php#L153](https://github.com/Wikia/app/blob/76ef81c3382df88f8708e5748c551f55ef556204/extensions/wikia/PhalanxII/fallback/Phalanx.class.php#L153): The Cyclomatic Complexity of this function "getFromFilterShort" is 21 which is greater than 20 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1541)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [Phalanx.class.php#L406](https://github.com/Wikia/app/blob/76ef81c3382df88f8708e5748c551f55ef556204/extensions/wikia/PhalanxII/fallback/Phalanx.class.php#L406): The Cyclomatic Complexity of this function "findBlocked" is 22 which is greater than 20 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1541)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [Phalanx.class.php#L457](https://github.com/Wikia/app/blob/76ef81c3382df88f8708e5748c551f55ef556204/extensions/wikia/PhalanxII/fallback/Phalanx.class.php#L457): Refactor this code to not nest more than 4 "if", "for", "while", "switch" and "try" statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS134)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [Phalanx.class.php#L472](https://github.com/Wikia/app/blob/76ef81c3382df88f8708e5748c551f55ef556204/extensions/wikia/PhalanxII/fallback/Phalanx.class.php#L472): Refactor this code to not nest more than 4 "if", "for", "while", "switch" and "try" statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS134)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [Phalanx.class.php#L483](https://github.com/Wikia/app/blob/76ef81c3382df88f8708e5748c551f55ef556204/extensions/wikia/PhalanxII/fallback/Phalanx.class.php#L483): Refactor this code to not nest more than 4 "if", "for", "while", "switch" and "try" statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS134)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [Phalanx.class.php#L77](https://github.com/Wikia/app/blob/76ef81c3382df88f8708e5748c551f55ef556204/extensions/wikia/PhalanxII/fallback/Phalanx.class.php#L77): 2 statements were found on this line. Reformat the code to have only one statement per line. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS122)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [Phalanx.class.php#L100](https://github.com/Wikia/app/blob/76ef81c3382df88f8708e5748c551f55ef556204/extensions/wikia/PhalanxII/fallback/Phalanx.class.php#L100): Define a constant instead of duplicating this literal "closestExpire" 8 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [Phalanx.class.php#L117](https://github.com/Wikia/app/blob/76ef81c3382df88f8708e5748c551f55ef556204/extensions/wikia/PhalanxII/fallback/Phalanx.class.php#L117): Define a constant instead of duplicating this literal "phalanx" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [Phalanx.class.php#L132](https://github.com/Wikia/app/blob/76ef81c3382df88f8708e5748c551f55ef556204/extensions/wikia/PhalanxII/fallback/Phalanx.class.php#L132): Define a constant instead of duplicating this literal "exact" 5 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)

ghost commented:
This will be getting another chance during Latin1→UTF-8 migration.

WikiaTech commented:
SonarQube analysis reported 23 issues
* ![MAJOR][MAJOR] 9 major
* ![MINOR][MINOR] 2 minor
* ![INFO][INFO] 12 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [Phalanx.class.php#L40](https://github.com/Wikia/app/blob/76ef81c3382df88f8708e5748c551f55ef556204/extensions/wikia/PhalanxII/fallback/Phalanx.class.php#L40): Replace this legacy wfMsg* function call with one of the new i18n functions. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3ALegacyWfMsgUsage)
1. ![MAJOR][MAJOR] [Phalanx.class.php#L77](https://github.com/Wikia/app/blob/76ef81c3382df88f8708e5748c551f55ef556204/extensions/wikia/PhalanxII/fallback/Phalanx.class.php#L77): Add curly braces around the nested statement(s). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS121)
1. ![MAJOR][MAJOR] [Phalanx.class.php#L153](https://github.com/Wikia/app/blob/76ef81c3382df88f8708e5748c551f55ef556204/extensions/wikia/PhalanxII/fallback/Phalanx.class.php#L153): The Cyclomatic Complexity of this function "getFromFilterShort" is 21 which is greater than 20 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1541)
1. ![MAJOR][MAJOR] [Phalanx.class.php#L268](https://github.com/Wikia/app/blob/76ef81c3382df88f8708e5748c551f55ef556204/extensions/wikia/PhalanxII/fallback/Phalanx.class.php#L268): Refactor this code to rely on RequestContext instead of context-dependent globals. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [Phalanx.class.php#L298](https://github.com/Wikia/app/blob/76ef81c3382df88f8708e5748c551f55ef556204/extensions/wikia/PhalanxII/fallback/Phalanx.class.php#L298): Replace or remove this deprecated Wikia::log logging statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AWikiaLogUsage)
1. ![MAJOR][MAJOR] [Phalanx.class.php#L406](https://github.com/Wikia/app/blob/76ef81c3382df88f8708e5748c551f55ef556204/extensions/wikia/PhalanxII/fallback/Phalanx.class.php#L406): The Cyclomatic Complexity of this function "findBlocked" is 22 which is greater than 20 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1541)
1. ![MAJOR][MAJOR] [Phalanx.class.php#L457](https://github.com/Wikia/app/blob/76ef81c3382df88f8708e5748c551f55ef556204/extensions/wikia/PhalanxII/fallback/Phalanx.class.php#L457): Refactor this code to not nest more than 4 "if", "for", "while", "switch" and "try" statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS134)
1. ![MAJOR][MAJOR] [Phalanx.class.php#L472](https://github.com/Wikia/app/blob/76ef81c3382df88f8708e5748c551f55ef556204/extensions/wikia/PhalanxII/fallback/Phalanx.class.php#L472): Refactor this code to not nest more than 4 "if", "for", "while", "switch" and "try" statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS134)
1. ![MAJOR][MAJOR] [Phalanx.class.php#L483](https://github.com/Wikia/app/blob/76ef81c3382df88f8708e5748c551f55ef556204/extensions/wikia/PhalanxII/fallback/Phalanx.class.php#L483): Refactor this code to not nest more than 4 "if", "for", "while", "switch" and "try" statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS134)
1. ![MINOR][MINOR] [Phalanx.class.php#L77](https://github.com/Wikia/app/blob/76ef81c3382df88f8708e5748c551f55ef556204/extensions/wikia/PhalanxII/fallback/Phalanx.class.php#L77): 2 statements were found on this line. Reformat the code to have only one statement per line. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS122)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'
